Documentation/building/build-tests-against-packages.md

```diff
@@ -0,0 +1,52 @@
+ - ```msbuild /p:BuildTestsAgainstPackages=true /p:SkipCleanPackages=true /p:PackagesDrops=[ProjectDir]bin/packages/[Relase|Debug]/ /t:BatchGenerateTestProjectJsons /t:BatchRestorePackages```
+  - /p:SkipCleanPackages=true is required for release branches where the packages folder is cleaned during every build.
+3. Build Tests against packages
+ - ```build-tests.cmd -BuildTestsAgainstPackages -- /p:"PackagesDrops=[ProjectDir]bin/packages/[Relase|Debug]/"```
+  - -BuildTestsAgainstPackages tells the build to use the project.json files you generated in the "Generate Test project.json files" step
+  - /p:"PackagesDrops=[ProjectDir]bin/packages/[Release|Debug]/" tells the build to use the packages from your local build drop.
+
+### Download product from an Azure blob
+
+This scenario skips the product build step, and instead downloads packages from Azure blob storage
+
+1. Sync product from Azure
+ - ```sync.cmd -ab -AzureAccount=dotnetbuildoutput -AzureToken=******** -Container=[Azure container name] -- /p:"DownloadDirectory=[ProjectDir]Packages\AzureTransfer" /p:"SkipCleanPackages=true"```
+2. Generate Test project.json files
+ - ```msbuild /p:BuildTestsAgainstPackages=true /p:SkipCleanPackages=true /p:PackagesDrops=[ProjectDir]Packages/AzureTransfer/[Relase|Debug]/ /t:BatchGenerateTestProjectJsons /t:BatchRestorePackages```
+  - /p:SkipCleanPackages=true is required for release branches where the packages folder is cleaned during every build.
+3. Build Tests against packages
+ - ```build-tests.cmd -BuildTestsAgainstPackages -- /p:"PackagesDrops=[ProjectDir]Packages/AzureTransfer/[Relase|Debug]/"```
+  - -BuildTestsAgainstPackages tells the build to use the project.json files you generated in the "Generate Test project.json files" step
+  - /p:"PackagesDrops=[ProjectDir]Packages/AzureTransfer/[Release|Debug]/" tells the build to use the packages from the Azure download (DownloadDirectory).
+
+### Use a versions file for specifying package versions
+
+This scenario uses a versions file (https://github.com/dotnet/versions/blob/master/build-info/dotnet/corefx/master/Latest_Packages.txt, for example) to determine what package versions to build tests against.
+
+1. Generate Test project.json files using a 'versions' file.
+ -   ```msbuild /p:BuildTestsAgainstPackages=true /p:SkipCleanPackages=true /p:VersionsFiles=[local version file path] /t:BatchGenerateTestProjectJsons /t:BatchRestorePackages```
+  - /p:SkipCleanPackages=true is required for release branches where the packages folder is cleaned during every build.
+2. Build Tests against packages
+ - ```build-tests.cmd -BuildTestsAgainstPackages -- /p:"PackagesDrops=[ProjectDir]bin/packages/[Relase|Debug]/"```
+  - -BuildTestsAgainstPackages tells the build to use the project.json files you generated in the "Generate Test project.json files" step
+  - /p:"PackagesDrops=[ProjectDir]bin/packages/[Release|Debug]/" tells the build to use the packages from your local build drop.
```

Comment:
If you are using the versions file wouldn't you be downloading all the packages from the feed? If so why do you need the PackagesDrops parameter?